packages/terra-dialog-modal/src/DialogModal.jsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import AbstractModal from 'terra-abstract-modal';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './DialogModal.scss';
+
+const cx = classNames.bind(styles);
+
+const widthFromSize = {
+  320: 320,
+  640: 640,
+  960: 960,
+  1280: 1280,
+  1600: 1600,
+};
+
+const propTypes = {
+ /**
+  * Aria Label of the dialog modal.
+  */
+  ariaLabel: PropTypes.string.isRequired,
+ /**
+  * Header of the dialog modal.
+  */
+  header: PropTypes.element.isRequired,
+ /**
+  * Footer of the dialog modal.
+  */
+  footer: PropTypes.element.isRequired,
+ /**
+  * Contents of the dialog modal.
+  */
+  children: PropTypes.node,
+  /*
+   * Callback function indicating a close condition was met.
+   */
+  onRequestClose: PropTypes.func.isRequired,
+  /*
+   * Toggle to show dialog modal or not.
+   */
+  isOpen: PropTypes.bool,
+  /*
+   * Toggle to focus on dialog modal or not.
+   */
+  isFocused: PropTypes.bool,
+  /**
+   * A callback function to let the containing component (e.g. modal) to regain focus.
+   */
+  releaseFocus: PropTypes.func,
+  /**
+   * A callback function to request focus from the containing component (e.g. modal).
+   */
+  requestFocus: PropTypes.func,
+  /**
+   * Width of the dialog modal. Defaults to 640px. Allows 320, 640, 960, 1280 and 1600.
+   */
+  width: PropTypes.oneOf(Object.keys(widthFromSize)),
+};
+
+const defaultProps = {
+  children: null,
+  isOpen: false,
+  isFocused: false,
+  releaseFocus: null,
+  requestFocus: null,
+  width: '640',
+};
+
+class DialogModal extends React.Component {
+  componentDidMount() {
+    if (this.props.isFocused && this.props.isOpen && this.props.requestFocus) {
+      this.props.requestFocus();
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.props.isOpen) {
+      if (this.props.requestFocus) {
+        this.props.requestFocus();
+      }
+    } else if (this.props.releaseFocus) {
+      this.props.releaseFocus();
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.props.releaseFocus) {
+      this.props.releaseFocus();
+    }
+  }
+
+  render() {
+    const { header,
+      footer,
+      children,
+      onRequestClose,
+      isOpen,
+      releaseFocus,
+      requestFocus,
+      ariaLabel,
+      isFocused,
+      width,
+      ...customProps
+    } = this.props;
+
+    if (!this.props.isOpen) {
+      return null;
+    }
+
+    return (
+      <AbstractModal
+        ariaLabel={this.props.ariaLabel}
+        role="dialog"
+        classNameModal={cx('dialog-modal-wrapper', `width-${widthFromSize[width]}`)}
```

Comment:
Found that this `width-${widthFromSize[width]}` doesn't check if `width` prop value is valid or not first, so if invalid value is given (i.e. '500' or 'taco'), the class `width-invalid` is applied to the element in the rendered markup and not falling back to default width as expected.